### PR TITLE
feat: add support for additionalProperties (#272)

### LINKED ISSUE #272: Cannot generate Maps via `additionalProperties`
#### Describe the bug

`additionalProperties` are not correctly mapped to a Map, instead they create an empty wrapper object `AnonymousSchemaX`.

#### How to Reproduce

Use a schema like

```yml
components:
  messages:
    metadata:
      payload:
        $id: SongMetaData
        type: object
        properties:
          tags:
            description: Tags
            additionalProperties:
              type: string
          stats:
            description: Stats
            additionalProperties:
              type: integer
              format: int64
          interprets:
            description: Interprets
            additionalProperties:
              $ref: '#/components/schemas/Interpret'
  schemas:
    Interpret:
      type: object
      properties:
        name:
          description: Interpret name
          type: string
```

#### Expected behavior

The previously mentioned schema should generate a class structure that can represent the following JSON:
```json
{
  "tags": {
    "tag1": "string",
    "tag2": "string"
  },
  "stats": {
    "stat1": 0,
    "stat2": 0
  },
  "interprets": {
    "some-key1": {
      "name": "string"
    },
    "some-key2": {
      "name": "string"
    }
  }
}
```

### LINKED ISSUE #322: Invalid Java class generated for free-form Map data type by use of 'additionalProperties' property
#### Describe the bug
Async Generator generates a invalid java class for free-form Map data type through 'additionalProperties' property in schema/model declarations.

#### How to Reproduce
a sample AsyncAPI looks like this
```
asyncapi: 2.6.0
info:
  title: Hello world application
  version: '0.1.0'
channels:
  hello:
    publish:
      message:
        payload:
          $ref: "#/components/schemas/keyValueType"

components:
  schemas:
    keyValueType:
      type: object
      additionalProperties: true
```

by use of AsyncApi/cli Generator like 
`asyncapi generate fromTemplate ./sample.yaml @asyncapi/java-spring-template -p javaPackage=Exporter.models --output myoutputTest`

an invalid java class is generated see its equals-method at last return, this class is not compilable

```
package Exporter.models.model;

import javax.validation.constraints.*;
import javax.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.List;
import java.util.Objects;

public class KeyValueType {  

    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        KeyValueType keyValueType = (KeyValueType) o;
        return 
    }

...
}

```

#### Expected behavior
a valid java class is generated for  'additionalProperties' property  to compile

### PR DESCRIPTION
**Pull request by JapuDCret**
**Description**

- Added the ability to convert from an object with `additionalProperties` to a `Map<String, type>`
- Created a test with different usages (String, Long, complex-type)

**Related issue(s)**

Resolves #272
Resolves #322

### CODE REVIEW

**Comment by Tenischev** [comment]
Hi @JapuDCret 
This looks cool!
Give me some time to do a full check.

File: filters/all.js [hunk 1]
```diff
@@ -2,7 +2,15 @@ const filter = module.exports;
 const _ = require('lodash');
 
 function defineType(prop, propName) {
-    if (prop.type() === 'object') {
+    if (prop.additionalProperties()) {
```
Comment (JapuDCret) [inline filters/all.js:5]: Note: Implicit check, if `prop.additionalProperties()` is either `false` or `undefined`/`null`

**Comment by JapuDCret** [comment]
Is there any update?

**Review by Tenischev — CHANGES_REQUESTED** [review]
I apologize for such an unacceptably long delay. Please consider suggested improvements

File: tests/mocks/map-format.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+servers:
+  production:
+    url: 'my-kafka-hostname:9092'
+    protocol: kafka
+    description: Production Instance 1
+channels:
+  song.metadata:
+    publish:
+      message:
+        $ref: '#/components/messages/metadata'
```
[the remark this answers is not in the document]
Comment (JapuDCret) [reply tests/mocks/map-format.yml:15]: I think $allOf is not supported on the `message` as per AsyncAPI definition.

I added a test-case for oneOf, though

**Review by Tenischev — CHANGES_REQUESTED** [review]

File: template/src/main/java/com/asyncapi/model/$$objectSchema$$.java [hunk 3]
```diff
@@ -17,7 +18,15 @@
 public class {{schemaName | camelCase | upperFirst}} {
     {% for propName, prop in schema.properties() %}
         {%- set isRequired = propName | isRequired(schema.required()) %}
-        {%- if prop.type() === 'object' %}
+        {%- if prop.additionalProperties() %}
+            {%- if prop.additionalProperties().type() === 'object' %}
```
[the remark this answers is not in the document]
Comment (Tenischev) [reply template/src/main/java/com/asyncapi/model/$$objectSchema$$.java]: fixed by me

File: filters/all.js (hunk 1 above)
Comment (Tenischev) [reply filters/all.js:5]: fixed by me

**Review by Tenischev — APPROVED** [review]

**Pull request merged**